### PR TITLE
Allow selectable sidebar calendar date and component updates

### LINKED ISSUE #76: Selecting date from sidebar calendar and display tasks for date
Currently, the sidebar calendar only displays the current date. The tasks list also only displays the current date's tasks. Create the following functionalities

---

**Sidebar Calendar**
- [x] Allow users to be able to **select a date from the sidebar calendar.**
    - Selected date should be highlighted in the middle
  
 
**Task List**
- [x] Display only tasks of the sidebar calendar's selected date. 
    - Show the date of the tasks in the header of the task list

_Sidebar Calendar And Task List For Reference: (the white rectangle is the selected date)_

image

### PR DESCRIPTION
**Pull request by EdwardLinS**
## Tracking Info

Resolves #76 

## Changes

- Allow two styles of selecting dates from the sidebar: scrolling and clicking
    - Scrolling increments and decrements the sidebar selected date
    - Clicking a date selects it as the selected date
 - Refactor to allow task list and completed statistics to listen to selected date changes using observer design pattern
 - Task list shows selected date
 - Change SQL query for completed statistics to count completed tasks for a specific date
 - Add end-to-end test for calendar scrolling and clicking selection methods.

## Testing

- Tested with the end-to-end test
- Verified with backend the correct tasks and completed statistics displayed
- Verified updating tasks changes loaded content

## Confirmation of Change

image

### CODE REVIEW

File: .github/workflows/jsdocs.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - staging
+      - edward/add-complete-delete-button-functionalities
```
Comment (dylantrann) [inline .github/workflows/jsdocs.yml:7]: should this be removed?

**Review by dylantrann — COMMENTED** [review]
Tested with Docker and works amazingly. 

Only issue is that when scrolling, it scrolls down the entire page BUT that's only and issue b/c the site is incomplete rn. Once everything is formatting correctly, the sit should fully encompass the window and scrolling won't matter

File: .github/workflows/jsdocs.yml (hunk 1 above)
Comment (EdwardLinS) [reply .github/workflows/jsdocs.yml:7]: Yes, forgot about that. Thanks!

**Review by dylantrann — APPROVED** [review]
Tested with `node server.js` and works as intended. 

Both scrolling and clicking on different dates work, and each date contain their specific tasks. Also checked the database in Docker to ensure that the entries were updated accordingly and located at their correct due dates.

**Pull request merged**